test/e2e/alertmanager_test.go

```diff
@@ -938,9 +939,10 @@ func testAlertmanagerConfigVersions(t *testing.T) {
 			Route: &monitoringv1alpha1.Route{
 				Receiver: "webhook",
 				Matchers: []monitoringv1alpha1.Matcher{{
-					Name:  "job",
-					Value: "webapp.+",
-					Regex: true,
+					Name:      "job",
+					MatchType: monitoringv1alpha1.MatchType(labels.MatchRegexp.String()),
```

Comment:
I wouldn't change the test: it verifies that the "legacy" API (e.g. when matchType is empty) works as expected.